clcache.py

```diff
@@ -1246,15 +1267,23 @@ def printStatistics(cache):
             stats.numCallsWithPch(),
         ))
 
+
 def resetStatistics(cache):
-    with cache.statistics as stats:
+    with cache.statistics.lock, cache.statistics as stats:
         stats.resetCounters()
     print('Statistics reset')
 
+
 def cleanCache(cache):
     with cache.statistics as stats, cache.configuration as cfg:
-        cache.clean(stats, cfg.maximumCacheSize())
-    print('Cache cleaned')
+        currentCacheSize = stats.currentCacheSize()
+        maximumCacheSize = cfg.maximumCacheSize()
+
+    if currentCacheSize >= maximumCacheSize:
+        with cache.lock, cache.statistics as stats:
+            cache.clean(stats, maximumCacheSize)
+        print('Cache cleaned')
```

Comment:
I moved the message printing out of `cleanCache` now (802ce00ba1d17710c705731088976) but I did not do the same change yet for `clearCache` and `resetStatistics`. I think it makes sense for consistency and symmetry, but it's probably worth a separate PR.